### PR TITLE
Create 404 template

### PR DESCRIPTION
**Pull request by dimtion**
Solve #430 

Do you think creating a method to generate a 404 template in the class pageBuilder is a good solution ?

### CODE REVIEW

**Comment by ArthurHoaro** [comment]
> Do you think creating a method to generate a 404 template in the class pageBuilder is a good solution ?

I would say yes, however there is an ongoing work to remove the pageBuilder class. See #397.

@virtualtam: What's your opinion on this? Should we anticipate this change here, or do you prefer to handle it in your PR?

Also, you only handle permalink 404, and the issue #430 mentions:

>  day not found (Daily page)
>  link not found
>  tag not found

But:
  * day not found in daily page display the current date.
  * tag not found shows a 'No results' search.

So I'm ok with it.

File: index.php [hunk 1]
```diff
@@ -681,6 +681,14 @@ public function renderPage($page)
         if ($this->tpl===false) $this->initialize(); // Lazy initialization
         $this->tpl->draw($page);
     }
+
+    // Render a 404 page (uses the template : tpl/404.tpl)
+    // e.g $PAGE->render404('The link is deleted')
```
Comment (ArthurHoaro) [inline index.php]: Use PHPDoc syntax.

File: tpl/404.html [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<!DOCTYPE html>
+<html>
+<head>{include="includes"}</head>
+<body onload="document.addform.post.focus();">
```
Comment (ArthurHoaro) [inline tpl/404.html]: No addform here (remove the `onload` attribute).

File: tpl/404.html [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+<div id="pageheader">
+	{include="page.header"}
+</div>
+<div class="error-container">
+    <h1>404 Not found <small>Oh crap!</small></h1>
```
Comment (ArthurHoaro) [inline tpl/404.html]: Indentation doesn't match the `include` one.

**Comment by dimtion** [comment]
>Also, you only handle permalink 404, and the issue #430 mentions:
>>    day not found (Daily page)
>>    link not found
> >   tag not found

> But:

>    * day not found in daily page display the current date.
>    * tag not found shows a 'No results' search.

> So I'm ok with it.

>Thanks for your work! 

Yeah, I wasn't able to reproduce a 404 for those pages so I waited for a feedback. However, thanks to `pageBuilder->render404()` it should not be too hard to create custom 404 pages for those.

**Comment by dimtion** [comment]
@virtualtam : would you prefer that I adapt and merge on your PR ?

**Comment by virtualtam** [comment]
@ArthurHoaro @dimtion I won't probably have time to work on #397 for a couple weeks (plus, it's become a bit obsolete & needs work), so don't worry too much about this, I'll take care of the necessary rebases & modifications when the time has come :)

**Comment by ArthurHoaro** [comment]
No problem. :)
@dimtion, can you [squash your commits](http://stackoverflow.com/questions/2563632/how-can-i-merge-two-commits-into-one)?

**Comment by dimtion** [comment]
Sure. :)

**Comment by ArthurHoaro** [comment]
Thanks!

**Pull request merged**